deployment/agent-deployment/ServiceGroupRoot/Scripts/pushAgentToAcr.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+set -e
+
+# Note - This script used in the pipeline as inline script
+
+if [ -z $AGENT_IMAGE_TAG_SUFFIX ]; then
+  echo "-e error value of AGENT_IMAGE_TAG_SUFFIX variable shouldnt be empty"
+  exit 1
+fi
+
+if [ ${#AGENT_IMAGE_TAG_SUFFIX} -ne 8 ]; then
+  echo "-e error length of AGENT_IMAGE_TAG_SUFFIX should be 8. Make sure it is in MMDDYYYY format"
+  exit 1
+fi
+
+if [ -z $AGENT_RELEASE ]; then
+  echo "-e error AGENT_RELEASE shouldnt be empty"
+  exit 1
+fi
+
+#Download agentimage tarball from blob storage account
+echo "Downloading tarball image from $AGENT_IMAGE_URI"
+wget -O $AGENT_IMAGE_TAR_FILE_NAME "${AGENT_IMAGE_URI}${WINDOWS}${RELEASE_ID}${AGENT_IMAGE_SAS}"
+
+
+if [ ! -f $AGENT_IMAGE_TAR_FILE_NAME ]; then
+    echo "Agent tarfile: ${AGENT_IMAGE_TAR_FILE_NAME} does not exist, unable to continue"
+    exit 1
+fi
+
+#Install crane
+echo "Installing crane"
+wget -O crane.tar.gz https://github.com/google/go-containerregistry/releases/download/v0.4.0/go-containerregistry_Linux_x86_64.tar.gz
+tar xzvf crane.tar.gz
+echo "Installed crane"
+
+
+#Login to az cli and authenticate to acr
+echo "Login cli using managed identity"
+az login --identity
+
+echo "Getting acr credentials"
+TOKEN_QUERY_RES=$(az acr login -n "$ACR_NAME" -t)
+TOKEN=$(echo "$TOKEN_QUERY_RES" | jq -r '.accessToken')
+DESTINATION_ACR=$(echo "$TOKEN_QUERY_RES" | jq -r '.loginServer')
```

Comment:
check if the TOKEN is empty, then fail the release with the appropriate error message.